src/echoclient/EchoClient.php

```diff
@@ -394,15 +398,20 @@ public static function setLogger(Logger $logger)
      * @param string $source
      * @param array $props
      * @param string|null $userId
+     * @param string|null $timestamp
      * @return string json encoded echo event
      */
-    protected function getEventJson($class, $source, array $props = array(), $userId = null)
+    protected function getEventJson($class, $source, array $props = array(), $userId = null, $timestamp = null)
```

Comment:
If my suggestion above were followed (have the old `createEvent` method construct an `EchoEvent` and call `createEvents`), this may not be required...?